### PR TITLE
#21473: Form element validation is not triggered when validation rules…

### PR DESCRIPTION
**Pull request by kisroman**
… change

### Fixed Issues (if relevant)

1. https://github.com/magento/magento2/issues/21473

### Manual testing scenarios (*)

1. Go to backend product edit page. Open console(F12).
2. Enter in description: "Some description". Write in console 
```
require('uiRegistry').get('index = description').setValidation('max_text_length', 2)
```
Now, you should see validation error under the description field.
3.  Set weight field empty. Write in console: 
```
require('uiRegistry').get('index = weight').setValidation('required-entry', true)
```
Now, you should see validation error under the description field.
 
### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @kisroman. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by rodrigowebjump** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by rodrigowebjump** [comment]
@magento-engcom-team give me test instanc

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump, here is your [Magento instance](https://i-21992-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-21992-2-3-develop.instances.magento-community.engineering/admin](https://i-21992-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by rodrigowebjump** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump, here is your [new Magento instance](https://pr-21992.instances.magento-community.engineering).
Admin access: [https://pr-21992.instances.magento-community.engineering/admin](https://pr-21992.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

File: app/code/Magento/Ui/view/base/web/js/form/element/abstract.js [hunk 1]
```diff
@@ -296,7 +296,7 @@ define([
                 this.validation[rule] = options;
             }
 
-            changed = utils.compare(rules, this.validation).equal;
+            changed = !(utils.compare(rules, this.validation).equal);
```
Comment (rodrigowebjump) [inline app/code/Magento/Ui/view/base/web/js/form/element/abstract.js]: Hi @kisroman 
Please see the Travis report - remove the unnecessary parentheses.

**Review by rodrigowebjump — CHANGES_REQUESTED** [review]

File: app/code/Magento/Ui/view/base/web/js/form/element/abstract.js (hunk 1 above)
Comment (kisroman) [reply app/code/Magento/Ui/view/base/web/js/form/element/abstract.js]: Done.

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-4923](https://jira.corp.magento.com/browse/ENGCOM-4923) has been created to process this Pull Request

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**